numbert/reranker.py

```diff
@@ -0,0 +1,237 @@
+'''
+Module for providing python interface to numBERT re-rankers
+'''
+
+import os
+os.environ["JAVA_HOME"] = "/usr/lib/jvm/java-11-openjdk-amd64"
+
+from pyserini.search import pysearch
+
+from .utils_reranker import load_model, online_eval, Hit
+from .scripts.utils_numbert import processors, output_modes, convert_examples_to_features
+
+class Reranker(object):
+    '''
+    Base class for Passage Ranking Models
+    (Defaults to the behavior of Anserini's search)
+    Parameters
+    ----------
+    index_dir : str
+        Path to Lucene index directory
+    '''
+    def __init__(self, index_dir):
+        self.simple_searcher = pysearch.SimpleSearcher(index_dir)
+
+    def rerank(self, q, k=100, t=-1, threads=1):
```

Comment:
Just a pep8 convention: no abreviated variables. E.g.: `q` becomes `query`, `k` becomes `num_hits`, etc..

Reply:
I was following the conventions of pyserini's [pysearch](https://github.com/castorini/pyserini/blob/master/pyserini/search/pysearch.py
), which does the same. I agree, this was not very descriptive, but I figured we might want to maintain some consistency across the two repositories.

Maybe @lintool can comment on this?

Reply:
I think _k_ is clear enough because top _k_ retrieval is a common convention. _t_ should be better documented in Pyserini.

Reply:
Since "t" is irrelevant here, I chose to ignore it so it will default to pysearch's t=-1, thereby ignoring the whole thing. I felt having many "k_i"s might confuse some users, so I instead renamed them num_hits_bm25, num_hits_mono, num_hits_duo.